riscv/triggers.cc

```diff
@@ -0,0 +1,200 @@
+#include "processor.h"
+#include "triggers.h"
+
+namespace triggers {
+
+mcontrol_t::mcontrol_t() :
+  type(2), maskmax(0), select(false), timing(false), chain_bit(false),
+  match(MATCH_EQUAL), m(false), h(false), s(false), u(false),
+  execute_bit(false), store_bit(false), load_bit(false)
+{
+}
+
+reg_t mcontrol_t::tdata1_read(const processor_t *proc) const noexcept {
```

Comment:
Could this be `const processor_t * const proc`? (Same question for many other `*proc` through this code)

Reply:
I find the notation for that pretty cumbersome, and the benefit negligible, so I generally don't bother.
But I'll add it.